lib/ansible/executor/task_executor.py

```diff
@@ -399,6 +396,34 @@ def _run_loop(self, items):
 
         return results
 
+    def _calculate_delegate_to(self, templar, variables):
+        if self._task.delegate_to:
+            delegated_host_name = templar.template(self._task.delegate_to, fail_on_undefined=False)
+            delegated_host = self._variable_manager._inventory.get_host(delegated_host_name)
```

Comment:
here instead of the 2 calls we should just try it as a pattern and ignore_restrictions/limits to start with

Reply:
Yeah, I didn't love the existing code, but it was a just a copy/paste from VM to here.  I didn't rewrite the code from scratch.